packages/medusa/src/commands/start.ts

```diff
@@ -120,21 +121,29 @@ async function start(args: {
     })
 
     try {
-      const { shutdown, gqlSchema, container } = await loaders({
+      const { shutdown, gqlSchema, container, modules } = await loaders({
         directory,
         expressApp: app,
       })
 
-      if (gqlSchema && generateTypes) {
-        const outputDirGeneratedTypes = path.join(directory, ".medusa/types")
-        await gqlSchemaToTypes({
-          outputDir: outputDirGeneratedTypes,
-          filename: "remote-query-entry-points",
-          interfaceName: "RemoteQueryEntryPoints",
-          schema: gqlSchema,
-          joinerConfigs: MedusaModule.getAllJoinerConfigs(),
+      if (generateTypes) {
+        await generateContainerTypes(modules, {
+          outputDir: path.join(directory, ".medusa/types"),
+          interfaceName: "ModuleImplementations",
         })
-        logger.info("Generated modules types")
+        logger.info("Generated container types")
+
+        if (gqlSchema) {
+          const outputDirGeneratedTypes = path.join(directory, ".medusa/types")
+          await gqlSchemaToTypes({
+            outputDir: outputDirGeneratedTypes,
+            filename: "remote-query-entry-points",
+            interfaceName: "RemoteQueryEntryPoints",
+            schema: gqlSchema,
+            joinerConfigs: MedusaModule.getAllJoinerConfigs(),
+          })
+          logger.info("Generated modules types")
```

Comment:
since: Since we write these on every server restart, these logs clutter things up a bit. Do you think we should remove them?

Reply:
Sure. That makes sense. I will switch from logger.info to logger.debug